pre_build/cccp_index_pre_build_check.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+"""Create jenkins jobs from the container-index."""
+
+import os
+import subprocess
+import sys
+import tempfile
+from glob import glob
+
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+jjb_defaults_file = 'pre-build-job.yml'
+
+required_attrs = ['image_name', 'image_version']
+optional_attrs = ['rundotshargs']
+overwritten_attrs = ['jobid', 'git_url', 'appid', 'jobs']
+
+
+def projectify(
+        new_project, appid, jobid, giturl, gitpath, gitbranch,
+        desiredtag, prebuild_script):
+
+    new_project[0]['project']['appid'] = appid
+    new_project[0]['project']['jobid'] = jobid
+    new_project[0]['project']['name'] = appid
+    new_project[0]['project']['git_url'] = giturl
+    new_project[0]['project']['git_branch'] = gitbranch
+    rel_path = "/"
+    if gitpath and gitpath != "/":
+        rel_path = gitpath if gitpath.startswith("/") else (rel_path + gitpath)
+    new_project[0]['project']['rel_path'] = rel_path
+    new_project[0]['project']['jobs'] = ['cccp-rundotsh-job']
+
+    if 'rundotshargs' not in new_project[0]:
+        new_project[0]['project']['rundotshargs'] = ''
+    elif new_project[0]['project']['rundotshargs'] is None:
+        new_project[0]['project']['rundotshargs'] = ''
```

Comment:
- In line # 35, we're checking for `rundotshargs` key in `new_project[0]`.
- In line # 37, we're checking for it in `new_project[0]['project']`. 

It's by design? I mean, in line # 35, we don't want to be doing `if 'rundotshargs' not in new_project[0]['project']`, right?

Reply:
these are unnecessary here, removed them.